src/Apitizer/Types/Filter.php

```diff
@@ -172,6 +156,13 @@ public function search($fields): self
      */
     protected function validateInput($input)
     {
+        if ($this->enums) {
+            if (!in_array($input, $this->enums)) {
+                throw InvalidInputException::filterTypeError($this, $input);
+            }
+            return TypeCaster::cast($input, $this->type, $this->format);
+        }
```

Comment:
The `TypeCaster` call doesn't need to be there, as it is handled later in the function, but more importantly, this will now fail for **any** array. So the following would still return an error:

```
$filter = $this->filter()->expect()->array()->whereEach()->enum(['blue', 'in', 'green']);
$filter->setValue(['blue', 'in']);
```

Even though all the given values in the array are correct enum values. So the rules are of the enum are as follows:
- If a single value is expected, that one value should be in the array of enums.
- If an array of values is expected, each of those elements should be a valid enum value.
- Otherwise, fail with `InvalidInputException`.

Reply:
Enum now follows these specified rules